### ISSUE TITLE
Included tasks in nested role has undefined variables

### ISSUE THREAD

**Issue by davidebelloni** [ansible/ansible#16320]
##### ISSUE TYPE
- Bug Report
##### ANSIBLE VERSION

```
ansible 2.1.0.0
```
##### OS / ENVIRONMENT

N/A
##### SUMMARY

Included tasks in nested role has undefined variables
##### STEPS TO REPRODUCE

In  ./inv

```
hostA ansible_connection=local

[some_group]
hostA
```

In ./playbook.yml

```
- hosts: hostA
  gather_facts: no
  roles:
    - role: roleA
      aaa: bbb
```

In ./roles/roleA/meta/main.yml

```
dependencies:
  - role: roleB
    ccc: "{{ aaa }}"
```

In ./roles/roleB/tasks/main.yml

```
- debug: msg={{ ccc }}
- include: sub_tasks.yml
```

In ./roles/roleB/tasks/sub_tasks.yml

```
- debug: msg="{{ ccc }}"
```
##### EXPECTED RESULTS

```
$ ansible-playbook -i inv playbook.yml 

PLAY [hostA] *******************************************************************

TASK [roleB : debug] ***********************************************************
ok: [hostA] => {
    "msg": "bbb"
}

TASK [roleB : include] *********************************************************
included: roles/roleB/tasks/sub_tasks.yml for hostA

TASK [roleB : debug] ***********************************************************
ok: [hostA] => {
    "msg": "bbb"
}

PLAY RECAP *********************************************************************
hostA                      : ok=3    changed=0    unreachable=0    failed=0
```
##### ACTUAL RESULTS

```
$ ansible-playbook -vvv -i inv playbook.yml 
Using ansible.cfg as config file

PLAYBOOK: playbook.yml *********************************************************
1 plays in playbook.yml

PLAY [hostA] *******************************************************************

TASK [roleB : debug] ***********************************************************
task path: roles/roleB/tasks/main.yml:1
ok: [hostA] => {
    "msg": "bbb"
}

TASK [roleB : include] *********************************************************
task path: roles/roleB/tasks/main.yml:2
included: roles/roleB/tasks/sub_tasks.yml for hostA

TASK [roleB : debug] ***********************************************************
task path: roles/roleB/tasks/sub_tasks.yml:1

fatal: [hostA]: FAILED! => {"failed": true, "msg": "{{ aaa }}: 'aaa' is undefined"}

PLAY RECAP *********************************************************************
hostA                      : ok=2    changed=0    unreachable=0    failed=1   
```

**Comment by ToBeReplaced** [comment]
I have the same issue `ansible-2.1.0.0-1.el7.noarch`.

This looks like the same issue as both #16080 and #16209, but I think these steps to reproduce are the simplest.

**Comment by jimi-c** [comment]
@davidebelloni this should be fixed in devel/stable-2.1 (included 2.1.1 rc1). Using your example above with `stable-2.1`:

<pre>
# ansible-playbook --version
ansible-playbook 2.1.1.0 (stable-2.1 a7199abb74) last updated 2016/06/22 08:33:08 (GMT -500)
  lib/ansible/modules/core: (detached HEAD b746f047ed) last updated 2016/06/22 19:00:05 (GMT -500)
  lib/ansible/modules/extras: (detached HEAD eae4134f96) last updated 2016/06/22 19:00:05 (GMT -500)
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/usr/share/ansible']
# ansible-playbook -vvv -i inv test.yml 
Using /etc/ansible/ansible.cfg as config file
PLAYBOOK: test.yml *************************************************************
1 plays in test.yml
PLAY [hostA] *******************************************************************
TASK [roleB : debug] ***********************************************************
task path: /root/testing/16320/roles/roleB/tasks/main.yml:3
ok: [hostA] => {
    "msg": "bbb"
}
TASK [roleB : debug] ***********************************************************
task path: /root/testing/16320/roles/roleB/tasks/sub_tasks.yml:1
ok: [hostA] => {
    "msg": "bbb"
}
PLAY RECAP *********************************************************************
hostA                      : ok=2    changed=0    unreachable=0    failed=0   
</pre>

As such, we're going to go ahead and close this now. If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular issue is resolved.

Thank you!
